### ISSUE TITLE
Duplicate Elasticsearch Init Container Env Variables

### ISSUE THREAD

**Issue by eoinburke67** [elastic/cloud-on-k8s#4382]
## Bug Report
**What did you do?**
Install ECK 1.2.1 & Elasticsearch 7.8

**What did you expect to see?**
A single definition of the Downward API env variables in the elastic init container

**What did you see instead? Under which circumstances?**
Duplicate entries for POD_NAME and POD_IP

**API Server Error**
E0315 14:31:42.570570         1 fieldmanager.go:159] [SHOULD NOT HAPPEN] failed to update managedFields for   /, Kind=: failed to convert new object (/v1, Kind=Pod) to smd typed: errors:       
.spec.initContainers[name="elastic-internal-init-filesystem"].env:   duplicate entries for key [name="POD_IP"]
.spec.initContainers[name="elastic-internal-init-filesystem"].env: duplicate entries for key [name="POD_NAME"]

```yaml
initContainers:
      - command:
        - bash
        - -c
        - /mnt/elastic-internal/scripts/prepare-fs.sh
        env:
        - name: POD_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - name: POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        - name: POD_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - name: POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
```

**Environment**
Giantswarm Release V13.0 on AWS
https://github.com/giantswarm/releases/blob/3e0ddd7b72054dc91d155f44630a3b43c4409df6/aws/v13.0.0/release.yaml
```yaml
  - name: aws-cni
    version: 1.7.6
  - name: calico
    version: 3.15.3
  - name: etcd
    version: 3.4.13
  - name: kubernetes
    version: 1.18.12
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.12", GitCommit:"7cd5e9086de8ae25d6a1514d0c87bac67ca4a481", GitTreeState:"clean", BuildDate:"2020-11-12T09:11:15Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**What I checked**

I started by looking code used to generate the init container since its definition is not in the helm charts 
https://github.com/elastic/cloud-on-k8s/blob/1.2.1/pkg/controller/elasticsearch/initcontainer/prepare_fs.go#L125

The variable are set here
https://github.com/elastic/cloud-on-k8s/blob/1.2.1/pkg/controller/common/defaults/pod_template.go#L17

And the defaults taken from 
https://github.com/elastic/cloud-on-k8s/blob/1.2.1/pkg/controller/elasticsearch/settings/environment.go#L18-L19

My Go knowledge is very basic but the above looks ok to me.

**Reproducibility**
At the moment I can reproduce this in all my 12 clusters.

**Comment by charith-elastic** [comment]
I am not familiar with the GiantSwarm distribution and it's not something we test against. Given that this problem does not happen on the Kubernetes distributions that we do test against, I am inclined to think that there might be a non-standard component in your installation that is trying to inject these variables automatically to new pods. For example, are there any mutating  webhooks that could be modifying the pods? (`kubectl get mutatingwebhookconfiguration`)

**Comment by eoinburke67** [comment]
Hi @charith-elastic,

```shell
> kubectl get mutatingwebhookconfiguration
NAME                          WEBHOOKS   AGE
cert-manager-webhook          1          194d
consul-connect-injector-cfg   1          31d
```
consul-connect-injector will inject a sidecar to a pod with specific annotations which Elasticsearch does not have. So nothing odd here.

I was checking with some other teams and they can reproduce this on other platforms so I dont think this is GiantSwarm related. We also noticed that when we deploy with eck 1.4 this issue is not reproducible.

**Comment by charith-elastic** [comment]
If you have a sidecar injector, you could be hitting this bug https://github.com/elastic/cloud-on-k8s/issues/3421. It was fixed in 1.2.1 though.

The current version of ECK is 1.5.0 and a lot of things have been fixed since the release of 1.2.1. If possible, please upgrade to 1.5.0 and let us know if the problem still persists.

**Comment by eoinburke67** [comment]
In this case there is no injection other than the elastic file system

```shell
> k get sts monitoring-elasticsearch-es-default -o json | jq '.spec.template.spec.initContainers[].name'
"elastic-internal-init-filesystem"
```
